.github/workflows/tag-build.yaml

```diff
@@ -1,10 +1,220 @@
+        
+          # Log the build details
+          echo "Built config:" | tee -a "$GITHUB_STEP_SUMMARY"
+          cat "$GITHUB_OUTPUT" | while read line; do
+            echo "* $line" | tee -a "$GITHUB_STEP_SUMMARY"
+          done
+      - name: ${{ inputs.workflow-tag }}
+        if: inputs.workflow-tag != ''
+        run: |
+          # Do nothing
+  # Each section here could be split out into a separate job, at the cost of slightly increased complexity.
+  # This would improve the (already somewhat fast) performance a bit, but I'm not sure if it's worth the
+  # tradeoff.
+  build-plugins:
+    needs:
+      - setup
+    runs-on: ubuntu-22.04-32core
+    environment: ${{ needs.setup.outputs.environment }}
+    permissions:
+      contents: read
+      id-token: write
+    env:
+      ARTIFACT_DIRECTORY: /tmp/build
+    steps:
+      # Setup
+      - name: Enable performance telemetry/metrics
+        uses: catchpoint/workflow-telemetry-action@v1
+        with:
+          comment_on_pr: false
+      - name: Checkout repo
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ needs.setup.outputs.gitref }}
+          fetch-depth: 0  # This is required by some of the commands in the makefiles
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: "./go.mod"
+          cache-dependency-path: "./go.sum"
+          check-latest: true
+      - name: Set environment variables for Makefiles
+        env:
+          VERSION_TAG: ${{ needs.setup.outputs.version }}
+        run: |
+          {
+            echo "VERSION=${VERSION_TAG##v}"
+            echo "GITREF=$VERSION_TAG" >> "$GITHUB_ENV"
+            echo "GNUMAKEFLAGS=-j$(nproc)"
+          } >> "$GITHUB_ENV"
+      # File artifacts
+      - name: Build the release tarballs
+        run: |
+          # Binaries and Helm charts
+          make releases helm-package-charts
+
+          # Terraform provider and event handler, as appropriate
+          go install github.com/konoui/lipo@latest  # At some point this should be merged into the buildbox
+          make OS=linux ARCH=amd64 release/terraform release/event-handler
+          make OS=linux ARCH=arm64 release/terraform
+          make OS=darwin ARCH=amd64 release/terraform release/event-handler
+          make OS=darwin ARCH=arm64 release/terraform
+          make OS=darwin ARCH=universal release/terraform
+      - name: Collect the build files
+        run: |
+          mkdir -pv "$ARTIFACT_DIRECTORY"
+          cp $(find . '(' -name "*.tar.gz" -o -name "*.tgz" ')' -type f) "$ARTIFACT_DIRECTORY/"
+      - name: Generate checksum files for built files
+        working-directory: ${{ env.ARTIFACT_DIRECTORY }}
+        run: |
+          find . '(' -name "*.tar.gz" -o -name "*.tgz" ')' -type f -exec sh -c 'sha256sum "$(basename {})" > "{}.sha256"' \;
```

Comment:
> shasum -a 256

I'm using `sha256sum` as this is the "standard" that we've unofficially picked across most of our repos and build processes. 

> works on both linux and mac

Is there an expectation that this workflow works on macos? If so then I need to go review everything again and look for other places where certain shell expressions work on Linux but not macos.

Also, you could make the same argument about Linux. `shasum` is usually packaged with `perl`, which is not available on all systems. However `coreutils` or `busybox` (which `sha256sum` is a part of) usually is.

> I'm still unsure why basename is needed.
As is `{}` contains the relative path to the file (i.e. `./some-tarball.tar.gz`). When passing this to `sha256sum` this value gets placed literally inside the checksum file. While this should be fine it doesn't match the format we've traditionally used for these files, and I _think_ I hit an issue somewhere downstream where the `./` wasn't interpreted properly.

Reply:
> Is there an expectation that this workflow works on macos? 

Not directly, but I've already faced this issue when doing the macOS workflows. Also this workflow does build macOS binaries, and while it currently does that on linux, if the source starts needing CGO, then this will need to run on a macOS builder and sha256sum was not installed on the mac builders. That's the only reason I suggested it.